### PR TITLE
Add Namespace field in prowjob

### PR DESCRIPTION
**Pull request by fejta**
Fixes https://github.com/kubernetes/test-infra/issues/10011

* Add a `Namespace string` field into `v1.ProwJobSpec`
* Add a `Namespace *string` field into `config.JobBase`
  - nil => `PodNamespace`
  - "" => `ProwJobNamespace`
* Ensure that (for now) this is always `PodNamespace` when `agent: kubernetes`
* Ensure that all the `validateFoo()` methods in `config` are unit tested (that's the majority of this PR)

/assign @cjwagner @stevekuznetsov 

Next PR will make the build controller honor `PodNamespace`

### CODE REVIEW

**Comment by fejta** [comment]
Anything I can do to get a review here?

**Comment by cjwagner** [comment]
> Add a Namespace *string field into config.JobBase
> - nil => PodNamespace
> - "" => ProwJobNamespace

This defaulting behavior seems very surprising. I don't think `nil ` should be treated differently from `""`. What is the rationale for defaulting to the ProwJobNamespace?

**Comment by fejta** [comment]
> I don't think nil should be treated differently from "". What is the rationale for defaulting to the ProwJobNamespace?

Done

Also the second commit is the only thing to review here. First commit is https://github.com/kubernetes/test-infra/pull/10125

File: prow/config/jobs.go [hunk 1]
```diff
@@ -91,6 +91,10 @@ type JobBase struct {
 	// Cluster is the alias of the cluster to run this job in.
 	// (Default: kube.DefaultClusterAlias)
 	Cluster string `json:"cluster,omitempty"`
+	// Namespace is the namespace in which pods schedule.
+	//   nil: results in config.PodNamespace (aka pod default)
+	//   empty: results in config.ProwJobNamespace (aka same as prowjob)
+	Namespace *string `json:"namespace,omitempty"`
```
Comment (cjwagner) [inline prow/config/jobs.go]: Add a comment to indicate that this field is not yet respected by the build controller.

File: config/jobs/kubernetes-security/generated-security-jobs.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ presubmits:
       preset-service-account: "true"
     max_concurrency: 12
     name: pull-security-kubernetes-e2e-kops-aws
+    namespace: test-pods
```
Comment (BenTheElder) [inline config/jobs/kubernetes-security/generated-security-jobs.yaml]: hmm. probably the security job generator should not write this field.

Comment (BenTheElder) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: can you add a line to this method to nil out the namespace? https://github.com/kubernetes/test-infra/blob/7c45489ea12f4b490ba1a890cda8beecc9d51a7e/config/jobs/kubernetes-security/genjobs.go#L123

**Pull request merged**